common/app/com/commercetools/sunrise/framework/viewmodels/forms/FormSettingsWithDefault.java

```diff
@@ -0,0 +1,21 @@
+package com.commercetools.sunrise.framework.viewmodels.forms;
+
+import play.mvc.Http;
+
+public interface FormSettingsWithDefault<T> extends FormSettings<T> {
+
+    T getDefaultValue();
+
+    /**
+     * Finds one selected valid value for this form in the HTTP request.
+     * If many valid values are selected, which one is going to be returned is non-deterministic.
+     * @param httpContext current HTTP context
+     * @return a valid selected value for this form, or the default value if no valid value is selected
+     */
+    default T getSelectedValueOrDefault(final Http.Context httpContext) {
```

Comment:
I really like how you changed the logic here: use `Optional<T>` when there is no default and use `T`if there is a default value 👍